control/descfcn.py

```diff
@@ -241,8 +241,8 @@ def describing_function_plot(
 
     """
     # Start by drawing a Nyquist curve
-    count, contour = nyquist_plot(
-        H, omega, plot=True, return_contour=True, **kwargs)
+    nyquist_plot(H, omega, **kwargs)
+    count, contour = frequency_response_nyquist(H, omega, **kwargs)
```

Comment:
This does the same calculation twice. Wasn't the whole point of the new paradigm to disentangle the calculation from the plotting? Ther must be some way to feed the result of `frequency_response_nyquist` into `nyquist_plot`.

Reply:
Thanks Ben. Regarding:
> Please keep your branch on top of current master. Step 3 and 8 in the wiki
I screwed up there. I should have read the wiki first... I forked the project to my own account and then branched it. Then I merged with my master pushed to remote and made the pull request from there! (and there are conflicts because my master became out of date).

Would you mind helping me reverse out of this situation?  Not surety best way to do that.  Maybe a quick offline chat if you have time some time?

Reply:
> This does the same calculation twice. Wasn't the whole point of the new paradigm to disentangle the calculation from the plotting?

Yes, I haven't implemented the new paradigm yet.  That is stage 2. 

I thought it would be better to separate the plotting and calculation functions as a first step (which also allows the [Object-Oriented approach to plotting in matplotlib](https://matplotlib.org/matplotblog/posts/pyplot-vs-object-oriented-interface/).

I thought this would be a good pre-cursor to implementing the object-oriented plotting paradigm.

Reply:
The beauty about git is that you can fix almost everything. Happy to help, although I am in Europe, so it has to be online :)

### Option 1

If you don't work on any other changes to python-control, you can keep on working with the basicmachines:master branch. Just don't consider it to be linked to python-control:master anymore.

```bash
git remote add upstream https://github.com/python-control/python-control.git
git fetch --all
git status
# make sure you are in a clean state, if not: commit or stash away pending changes: git stash
git checkout upstream/master -b control-master
# from here on, follow the wiki but replace master with control-master and <my-new-branch-name> with master
git checkout master
git rebase control-master
# if you stashed before: git stash pop
```

You are at step 11 now.

### Option 2
If you want to change the basicmachines:master branch, you have to close this PR and open a new one:

```
git remote add upstream https://github.com/python-control/python-control.git
git fetch --all
git status
# make sure you are in a clean state, if not: commit or stash away pending changes: git stash
git checkout -b new-plotting-paradigm
git checkout master
git reset --hard upstream/master
# Make sure #689 is closed before next step
git push -f
git checkout new-plotting-paradigm
git rebase master
# if you stashed before: git stash pop
git push --set-upstream origin new-plotting-paradigm
```

Continue with step 10.